lib/database/player-game.php

```diff
@@ -275,17 +275,21 @@ function getUsersCompletedGamesAndMax($user): array
     $requiredFlags = AchievementType::OfficialCore;
     $minAchievementsForCompletion = 5;
 
-    $query = "SELECT gd.ID AS GameID, c.Name AS ConsoleName, c.ID AS ConsoleID, gd.ImageIcon, gd.Title, COUNT(ach.GameID) AS NumAwarded, inner1.MaxPossible, (COUNT(ach.GameID)/inner1.MaxPossible) AS PctWon, aw.HardcoreMode
+    $query = "SELECT
+            gd.ID AS GameID, c.Name AS ConsoleName, c.ID AS ConsoleID, gd.ImageIcon, gd.Title, inner1.MaxPossible,
+            SUM(aw.HardcoreMode = 0) AS NumAwarded, SUM(aw.HardcoreMode = 1) AS NumAwardedHC, 
+            (SUM(aw.HardcoreMode = 0) / inner1.MaxPossible) AS PctWon,
+            (SUM(aw.HardcoreMode = 1) / inner1.MaxPossible) AS PctWonHC, aw.HardcoreMode
         FROM Awarded AS aw
         LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID
         LEFT JOIN GameData AS gd ON gd.ID = ach.GameID
         LEFT JOIN
             ( SELECT COUNT(*) AS MaxPossible, ach1.GameID FROM Achievements AS ach1 WHERE Flags = $requiredFlags GROUP BY GameID )
             AS inner1 ON inner1.GameID = ach.GameID AND inner1.MaxPossible > $minAchievementsForCompletion
         LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
-        WHERE aw.User='$user' AND ach.Flags = $requiredFlags
-        GROUP BY ach.GameID, aw.HardcoreMode, gd.Title
-        ORDER BY PctWon DESC, inner1.MaxPossible DESC, gd.Title ";
+        WHERE aw.User = '$user' AND ach.Flags = $requiredFlags
+        GROUP BY ach.GameID, gd.Title
+        ORDER BY PctWon = 1.0 DESC, PctWonHC DESC, PctWon DESC, inner1.MaxPossible DESC, gd.Title";
```

Comment:
Something about this syntax is not supported by the xampp enviroment.
```
MariaDB [retroachievements-web]> SELECT gd.ID AS GameID, c.Name AS ConsoleName, c.ID AS ConsoleID, gd.ImageIcon, gd.Title, inner1.MaxPossible, SUM(aw.HardcoreMode = 0) AS NumAwarded, SUM(aw.HardcoreMode = 1) AS NumAwardedHC, (SUM(aw.HardcoreMode = 0) / inner1.MaxPossible) AS PctWon, (SUM(aw.HardcoreMode = 1) / inner1.MaxPossible) AS PctWonHC, aw.HardcoreMode FROM Awarded AS aw LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID LEFT JOIN GameData AS gd ON gd.ID = ach.GameID LEFT JOIN ( SELECT COUNT(*) AS MaxPossible, ach1.GameID FROM Achievements AS ach1 WHERE Flags = 3 GROUP BY GameID ) AS inner1 ON inner1.GameID = ach.GameID AND inner1.MaxPossible > 5 LEFT JOIN Console AS c ON c.ID = gd.ConsoleID WHERE aw.User = 'Jamiras' AND ach.Flags = 3 GROUP BY ach.GameID, gd.Title ORDER BY PctWon = 1.0 DESC, PctWonHC DESC, PctWon DESC, inner1.MaxPossible DESC, gd.Title;
ERROR 1247 (42S22): Reference 'PctWon' not supported (reference to group function)
```

Reply:
From what I could gather here, this seems to be a specific DB/engine limitation caused by an alias expression being used in `ORDER BY`. Try replacing `PctWon = 1.0` for `(SUM(aw.HardcoreMode = 0) / inner1.MaxPossible) = 1.0` and see if the issue is solved for you.

Reply:
That does seem to work. Thanks.